.github/workflow-scripts/label-products.js

```diff
@@ -58,30 +58,41 @@ async function run() {
 async function labelPRSubFolders(octokit, repo, prNumber, changedFolders) {
   const labelPrefix = 'product:';
   const labelsToRemove = [];
+  const labelsToAdd = [];
 
   for (const label of github.context.payload.pull_request.labels) {
     if (label.name.startsWith(labelPrefix)) {
       labelsToRemove.push(label.name);
     }
   }
 
-  for (const labelToRemove of labelsToRemove) {
-    await octokit.rest.issues.removeLabel({
-      ...repo,
-      issue_number: prNumber,
-      name: labelToRemove
-    });
-  }
-
   for (const folder of changedFolders) {
     const label = labelPrefix + folder;
+    labelsToAdd.push(label);
+  }
 
-    await octokit.rest.issues.addLabels({
-      ...repo,
-      issue_number: prNumber,
-      labels: [label]
-    });
+  const currentLabels = new Set(github.context.payload.pull_request.labels.map(label => label.name));
+
+  for (const labelToRemove of labelsToRemove) {
+    if (!labelsToAdd.includes(labelToRemove)) {
+      await octokit.rest.issues.removeLabel({
+        ...repo,
+        issue_number: prNumber,
+        name: labelToRemove
+      });
+    }
+  }
+
+  for (const labelToAdd of labelsToAdd) {
+    if (!currentLabels.has(labelToAdd)) {
+      await octokit.rest.issues.addLabels({
+        ...repo,
+        issue_number: prNumber,
+        labels: [labelToAdd]
+      });
+    }
   }
```

Comment:
```suggestion
  let newLabels = [];
  for (const labelToAdd of labelsToAdd) {
    if (!currentLabels.has(labelToAdd)) {
      newLabels.push(labelToAdd);
    }
  }
  
  if (newLabels.length > 0) {
    await octokit.rest.issues.addLabels({
      ...repo,
      issue_number: prNumber,
      labels: newLabels
    });
  }
```
Nit: We could call the `addLabels` method only once, since it accepts an array of labels. (I didn't test this change!)